book/src/leader-rotation.md

```diff
@@ -3,17 +3,92 @@
 packets, the cluster cannot simply elect a single node to hold the leader role
-indefinitely. Instead, the cluster minimizes the influence of a malcioius
+indefinitely. Instead, the cluster minimizes the influence of a malicious
 leader by rotating which node takes the lead.
 
 Each validator selects the expected leader using the same algorithm, described
 below. When the validator receives a new signed ledger entry, it can be certain
-that entry was produced by the expected leader.
+that entry was produced by the expected leader.  The order of slots which each
+leader is assigned a slot is called a *leader schedule*.
 
-## Leader Schedule Generation
+## Leader Schedule Rotation
+
+A validator rejects blocks that are not signed by the *slot leader*.  The list
+of identities of all slot leaders is called a *leader schedule*. The leader
+schedule is recomputed locally and periodically. It assigns slot leaders for a
+duration of time called an _epoch_. The schedule must be computed far in advance
+of the slots it assigns, such that the ledger state it uses to compute the
+schedule is finalized. That duration is called the *leader schedule offset*.
+Solana sets the offset to the duration of slots until the next epoch. That is,
+the leader schedule for an epoch is calculated from the ledger state at the
+start of the previous epoch. The offset of one epoch is fairly arbitrary and
+assumed to be sufficiently long such that all validators will have finalized
+their ledger state before the next schedule is generated. A cluster may choose
+to shorten the offset to reduce the time between stake changes and leader
+schedule updates.
+
+While operating without partitions lasting longer than an epoch the schedule
+only needs to be generated when the root fork crosses the epoch boundary.  Since
+the schedule is for the next epoch, any new stakes committed to the root fork
+will not be active until the next epoch.
+
+Without a partition lasting longer then an epoch, the cluster will work as
+follows:
+
+1. A validator continuously updates its own root fork as it votes.
+
+2. The validator updates its leader schedule each time the slot height crosses
+an epoch boundary.
+
+For example:
+
+The epoch duration is 100 slots. The root fork is updated from fork computed at
+slot height 99 to a fork computed at slot height 102. Forks with slots at height
+100,101 were skipped because of failures.  The new leader schedule is computed
+using fork at slot height 102.  It is active from slot 200 until it is updated
+again.
+
+No inconsistency can exist because every validator that is voting with the
+cluster has skipped 100 and 101 when its root reaches 102.  The duration of the
+*leader schedule offset* has a direct relationship to the likelihood of a
+cluster inconsistency state.  If the *leader schedule offset* is 6 standard
+deviations longer than the median partition time, then the likelihood of a
+partition longer than the offset is 1 in 1 million.
```

Comment:
I'm not able to connect the dots on this math. Can you spell it out for me?

Reply:
@garious that is the definition of standard deviation.